benches/big_tree.rs

```diff
@@ -0,0 +1,68 @@
+//! This file includes benchmarks for very large, pseudo-randomly generated trees
+use criterion::{criterion_group, criterion_main, Criterion};
+use rand::prelude::*;
+use rand_chacha::ChaCha8Rng;
+use taffy::prelude::*;
+use taffy::style::FlexboxLayout;
+
+/// The number of nodes to include in the trees
+const NODE_COUNT: u32 = 100_000;
+
+/// Get a randomly generated dimension
+fn get_random_dimension(rng: &mut ChaCha8Rng) -> Dimension {
+    let switch: f32 = rng.gen_range(0.0..=1.0);
+
+    match switch {
+        0.0..=0.2 => Dimension::Auto,
+        0.2..=0.4 => Dimension::Undefined,
+        0.4..=0.8 => Dimension::Points(rng.gen_range(0.0..500.0)),
+        _ => Dimension::Percent(rng.gen_range(0.0..1.0)),
+    }
+}
+
+/// Get a randomly generated size
+fn get_random_size(rng: &mut ChaCha8Rng) -> Size<Dimension> {
+    Size { width: get_random_dimension(rng), height: get_random_dimension(rng) }
+}
+
+/// Get a randomly generated style for a node
+fn get_random_style(rng: &mut ChaCha8Rng) -> FlexboxLayout {
```

Comment:
IMO this should be a method on `FlexboxLayout`. I would actually put this into the main crate and hide it behind a off-by-default feature flag. This functionality would be excellent for downstream crates like bevy and dioxus as well.

I think the best architecture here is actually a `Random` trait that we can implement for each of the fields on the `FlexboxLayout`, as well as on the `FlexboxLayout` itself.

Reply:
How would a random layout style be useful for `bevy` and `dioxus`? Or do you mean for their benchmarks?

This would require us to add the rng dependencies to the main crate, but I guess if it's behind a feature flag that's okay.

Reply:
One problem of this approach might be that you might want to fine-tune the probabilities of each style.

Another problem might be that for deep layouts, it's easy to create nonsensical styles, e.g. a parent node will often be smaller than the child nodes. I'm not sure what the best way would be to fix this.

Reply:
> How would a random layout style be useful for bevy and dioxus? Or do you mean for their benchmarks?

For their benchmarks, yes :)

Reply:
@alice-i-cecile One problem with this approach would be that I can not find a way to enable specific features by default only in a given profile.

We don't want to have the `rand` feature enabled by default, but we do want it in the `bench` profile.

I guess one way to resolve this would be to explicitly enable the feature with the CLI flag and to only define the benchmarks if the feature is enabled. However, then you would need to always remember to enable the feature.

Do you know of a better way to do this?

Reply:
Hmm. I think if taffy defines these methods in a feature flag, downstream dependencies will be able to use that feature flag in their dev dependencies?

I think that having to remember to add a flag to the dev dependencies a single time per library is fine.

Reply:
And how can we handle that for our own benchmarks? I haven't found a way to enable a feature for a specific profile only.

Reply:
We should be able to import `taffy` as a dev-dependency within this crate, and can enable additional features there.